pkg/client/restclient/plugin.go

```diff
@@ -30,9 +30,16 @@ type AuthProvider interface {
 	// WrapTransport allows the plugin to create a modified RoundTripper that
 	// attaches authorization headers (or other info) to requests.
 	WrapTransport(http.RoundTripper) http.RoundTripper
+	// Login is expected to maybe require user interaction; it won't be called
+	// in environments where a user is not present
+	Login() error
```

Comment:
> Neither @bobbyrullo's use case or mine would need this right now. Can we defer on adding this to the plugin until we have a user?

If you update the comment to say that it "must not require user interaction", then ok.